normalize.js

```diff
@@ -380,10 +374,49 @@ var buildCustomSchema = exports.buildCustomSchema = function (schema, types, par
           fields[field.uid] = '' + blockparent;
         }
         break;
+      case 'reference':
+        var unionType = 'union ';
+        if (typeof field.reference_to === 'string' || field.reference_to.length === 1) {
+          field.reference_to = Array.isArray(field.reference_to) ? field.reference_to[0] : field.reference_to;
+          var _type2 = 'type ' + prefix + '_' + field.reference_to + ' implements Node { title: String! }';
+          types.push(_type2);
+          if (field.mandatory) {
+            fields[field.uid] = '[' + prefix + '_' + field.reference_to + ']!';
+          } else {
+            fields[field.uid] = '[' + prefix + '_' + field.reference_to + ']';
+          }
+        } else {
+          var unions = [];
+          field.reference_to.forEach(function (reference) {
+            var referenceType = prefix + '_' + reference;
+            unionType = unionType.concat(referenceType);
+            unions.push(referenceType);
+            var type = 'type ' + referenceType + ' implements Node { title: String! }';
+            types.push(type);
+          });
+          var name = '';
+          name = name.concat(unions.join(''), '_Union');
+          unionType = unionType.concat('_Union = ', unions.join(' | '));
+          types.push(unionType);
+
+          references.push({
```

Comment:
i believe just pushing an object with just a parent and a uid is not sufficient to resolve queries afterwards, as demonstrated by the runtime error that is looking for an internal field with a type

Reply:
while debugging, i noticed the related_to field being resolved by the default resolver and not the custom resolver we register...

Reply:
it looks like not all reference fields get our custom resolvers registered, but some fallback to the default resolver, with the runtime error as consequence. Not sure why yet...